api/src/main/scala/dx/api/DxInstanceType.scala

```diff
@@ -524,29 +537,28 @@ object InstanceTypeDB extends DefaultJsonProtocol {
             )
         )
     )
-    val (availableInstanceTypes, pricingAvailable) =
-      try {
-        val pricingModel = getPricingModel(api, projectDesc.billTo.get, projectDesc.region.get)
-        val instanceTypesWithPrices = allInstanceTypes.keySet
-          .intersect(pricingModel.keySet)
-          .map { name =>
-            name -> allInstanceTypes(name).copy(price = pricingModel.get(name))
-          }
-          .toMap
-        (instanceTypesWithPrices, true)
-      } catch {
-        case ex: Throwable =>
-          logger.warning(
-              """|Warning: insufficient permissions to retrieve the
-                 |instance price list. This will result in suboptimal machine choices,
-                 |incurring higher costs when running workflows.
-                 |""".stripMargin.replaceAll("\n", " ")
-          )
-          (allInstanceTypes, false)
+    val availableInstanceTypes =
+      getPricingModel(api, projectDesc.billTo.get, projectDesc.region.get)
+        .map { pricingModel =>
```

Comment:
I was wondering if calling map() on an Option is ok, but it looks like it is

```
Expression type:

Option[Map[String,DxInstanceType]]
Symbol signature:

final def map[B](f: Map[String,Float] => B): Option[B]
Returns a scala.Some containing the result of applying f to this scala.Option's value if this scala.Option is nonempty. Otherwise return None.

This is equivalent to:

option match {
  case Some(x) => Some(f(x))
  case None    => None
}
```

Reply:
yes, it must means "perform this operation if the option is not empty"